packages/@orbit/validators/src/object-validator.ts

```diff
@@ -0,0 +1,35 @@
+import { ValidationIssue, ValidationOptions, Validator } from './validator';
+
+interface BaseIssue extends ValidationIssue<unknown> {
+  validator: 'object';
+}
+
+interface TypeIssue extends BaseIssue {
+  validation: 'type';
+}
+export type ObjectValidationIssue = TypeIssue;
+
+export type ObjectValidationOptions = ValidationOptions;
+
+export type ObjectValidator = Validator<
+  unknown,
+  ObjectValidationOptions,
+  ObjectValidationIssue
+>;
+
+export const validateObject: ObjectValidator = (
+  input: unknown,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  options?: ObjectValidationOptions
+): undefined | ObjectValidationIssue[] => {
+  if (typeof input !== 'object') {
```

Comment:
`typeof null === 'object'` fyi @dgeb

Reply:
@joelgallant - ah good catch! This won't have practical implications when used to validate attributes of type `object` because `null` [will be caught first](https://github.com/orbitjs/orbit/blob/main/packages/@orbit/records/src/record-validators/record-attribute-validator.ts#L120). 

Regardless, the object validator should work well on its own and should probably not accept `null`.

Reply:
Tracking this in #859